### ISSUE TITLE
[Question]: Interested only in final sql statement

### ISSUE THREAD

**Issue by vp999** [jerryjliu/llama_index#7337]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

i am using NLSQLTableQueryEngine with llamacpp as backend  on azure notebook.  The query_engine.query() returns results as well as sql statement. how do i restrict it to return only final sql and not to execute it?

**Comment by vp999** [comment]
Shouldn’t there a be a feature that allows post processing every  sql query before executing it on database ?  This will help to add any custom logic like data authorisation, additional changes specific to sql dialect..

**Comment by logan-markewich** [comment]
@vp999 yea the bot is wrong

Simply set `synthesize_response=False` in the kwargs

`NLSQLTableQueryEngine(..., synthesize_response=False)`

source: https://github.com/jerryjliu/llama_index/blob/8611c2f0f2a53e4d46f8d76d7be7485077bb206f/llama_index/indices/struct_store/sql_query.py#L326

**Comment by vp999** [comment]
synthesize_response=False doesnt help .. it  still executes the sql ..

**Comment by logan-markewich** [comment]
Oh whoops, my brain went to another thought (stopping the natural language generation)

Turns out the bot was smarter than me. There's no option for it currently. Would be welcome to having a PR though!

**Comment by logan-markewich** [comment]
The sql will always get ran currently
https://github.com/jerryjliu/llama_index/blob/8611c2f0f2a53e4d46f8d76d7be7485077bb206f/llama_index/indices/struct_store/sql_query.py#L280

**Comment by vp999** [comment]
Yes .. thats where we need a hook for modifying the sql before executing ..
